lib/string_reverse.rb

```diff
@@ -1,6 +1,17 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  return nil if my_string.nil?
+  n = my_string.length
+  i = 0
+  j = n - i - 1
+
+  while i < j
+    word = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = word

```

Comment:
If you wanted, you could do this on one line:

```ruby
my_string[i], my_string[j] = my_string[j], my_string[i]
```

but it's just as valid to use a temporary variable.  I wouldn't call that variable `word`, though, since it isn't holding a word.